app/league/all/page.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import Leagues from './page';
+
+jest.mock('@/store/dataStore', () => ({
+  useDataStore: jest.fn(),
+}));
+
+jest.mock('@/utils/utils', () => ({
+  getUserLeagues: jest.fn(),
+}));
+
+jest.mock('@/api/apiFunctions', () => ({
+  getGameWeek: jest.fn(),
+}));
+
+describe('Leagues Component', () => {
+  const mockUser = { id: '123', leagues: [] };
+  const mockUseDataStore = require('@/store/dataStore').useDataStore;
```

Comment:
Why is mocking the function like this?

Reply:
To make the test more accessible, but let me take a look to revise them.